src/tmx/TmxApi/tmx/messages/TmxJ2735Codec.hpp

```diff
@@ -358,11 +358,10 @@ class TmxJ2735EncodedMessage: public TmxJ2735EncodedMessageBase
 			else if (is_uper())
 			{
 				if (msgId > MessageFrameMessage::get_default_messageId())
-				{
-					MessageFrameMessage *frame = TmxJ2735EncodedMessage<MessageFrameMessage>::decode_j2735_message<
-							codec::uper<MessageFrameMessage> >(theData);
-					if (frame)
-						_decoded.reset(new MsgType(frame->get_j2735_data()));
+				{	
+					_frame.reset(TmxJ2735EncodedMessage<MessageFrameMessage>::decode_j2735_message<
```

Comment:
where is this `_frame `initialized? Do we need to keep the check on `_frame `pointer as well to ensure it is not nullptr?